### ISSUE TITLE
Make heading-offset cumulative over nested inclusions

### ISSUE THREAD

**Issue by mirk8s** [mondeja/mkdocs-include-markdown-plugin#52]
Use case is:

**File A.md**: 
```# A``` 

**File B.md**: 
```# B```

**File C.md**: 
```
#C

{%
include-markdown "./A.md"
heading-offset=1
%}

{%
include-markdown "./B.md"
heading-offset=1
%}
```
which is already rendered as following:

# C

## B

But if you have
**File D.md**: 
```
# D

{%
include-markdown "./C.md"
heading-offset=1
%}
```

it is currently rendered as:

# D

## # B

While I expected it to be rendered as follows:

# D

### B

Therefore, I suggest making **heading-offset** parameter cumulative over nested inclusions in order to achieve that expected result. Thank you for your feedback.

**Comment by mondeja** [comment]
Added in [v3.1.1](https://github.com/mondeja/mkdocs-include-markdown-plugin/releases/tag/v3.1.1).

**Comment by mirk8s** [comment]
Thank you for your promptitude on yesterday release, it's working just fine for straightforward cases.
However I still have this more complex case which isn't working as expected.

**File 4.md**:
```
#### 4

{%
include-markdown "./4.1.md"
heading-offset=4
%}

{%
include-markdown "./4.2.md"
heading-offset=4
%}
```

**File 4.1.md**:
```
# 4.1

{%
include-markdown "./4.1.1.md"
heading-offset=1
%}

{%
include-markdown "./4.1.2.md"
heading-offset=1
%}
```

**File 4.1.1.md**:
```
# 4.1.1

{%
include-markdown "./4.1.1.1.md"
heading-offset=1
%}

{%
include-markdown "./4.1.1.2.md"
heading-offset=1
%}
```

**File 4.1.1.1.md**:
```
# 4.1.1.1


```

**File 4.1.1.2.md**:
```
# 4.1.1.2


```

**File 4.1.2.md**:
```
# 4.1.2


```

**File 4.2.md**:
```
# 4.2

{%
include-markdown "./4.2.1.md"
heading-offset=1
%}

{%
include-markdown "./4.2.2.md"
heading-offset=1
%}
```

**File 4.2.1.md**:
```
# 4.2.1


```

**File 4.2.2.md**:
```
# 4.2.2


```

## EXPECTED

#### 4
##### 4.1
###### 4.1.1
###### # 4.1.1.1
###### # 4.1.1.2
###### 4.1.2
##### 4.2
###### 4.2.1
###### 4.2.1

## ACTUAL

#### 4
##### 4.1
###### ### 4.1.1
###### ######### 4.1.1.1
###### ######### 4.1.1.2
###### ### 4.1.2
##### 4.2
###### ### 4.2.1
###### ### 4.2.1

At first sight, it seems that in nested inclusions (with different heading-offset) parent heading-offset values are added up more times, while each parent heading-offset value should be added only once.
I try to provide a schema of the algorithm as I expect it to work. Sorry but I don't feel confident making a PR since I never used Python.

| File         | Parent Incoming Offset  |   | Previous Incoming Offsets Sum |   | Self Heading | Self Result | Child Outgoing Offset
| -------------| ------------------------|---|-------------------------------|---|--------------|-------------|----------------------
| 4.1.md 	   |	4# 	                 | + |    				             | + |    1#        | = 5#		  | 1#
| 4.1.1.md 	   |	1# 	                 | + | 4# 				           	 | + |    1#        | = 6#		  | 1#
| 4.1.1.1.md   |	1# 	                 | + | 4# + 1# 			             | + |    1#        | = 7#        |

**Comment by mondeja** [comment]
Ciuld you provide a minimal repository example with these files?

**Comment by mondeja** [comment]
Thanks for the report! Should have been fixed in [v3.1.4](https://github.com/mondeja/mkdocs-include-markdown-plugin/releases/tag/v3.1.4). Could you check it?

**Comment by mirk8s** [comment]
Yes it works as expected, thanks!
